### PR TITLE
#471 adapt generator par following os 2221

### PR DESCRIPTION
**Pull request by marcochiaramello**
**Checklist before requesting a review**

*use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes*

- [x] unit tests and non regression tests were added (update of inputs, outputs, SA, algos)
- [ ] main documentation was updated (update of input/output file)
- [ ] if this commit targets a release branch: the corresponding milestone was added in the ticket and in this PR

### CODE REVIEW

**Comment by marcochiaramello** [comment]
This follows OS 2221: https://github.com/dynawo/dynawo/pull/2545

**Comment by marcochiaramello** [comment]
Closes #487

File: sources/Outputs/include/ParGenerator.h [hunk 1]
```diff
@@ -31,6 +32,9 @@ namespace outputs {
  */
 class ParGenerator {
  public:
+  using ActivePowerCompensation = dfl::inputs::Configuration::ActivePowerCompensation;  ///< Alias for type of active power compensation
+  using StartingPointMode = dfl::inputs::Configuration::StartingPointMode;              ///< Alias for strating point mode
```
Comment (rosiereflo) [inline sources/Outputs/include/ParGenerator.h:36]: typo: starting

File: sources/Outputs/include/ParGenerator.h [hunk 2]
```diff
@@ -48,12 +52,9 @@ class ParGenerator {
    * @param busesWithDynamicModel map of bus ids to a generator that regulates them
    * @param startingPointMode starting point mode
    */
-  void write(boost::shared_ptr<parameters::ParametersSetCollection>& paramSetCollection,
-             dfl::inputs::Configuration::ActivePowerCompensation activePowerCompensation,
-             const std::string& basename,
-             const boost::filesystem::path& dirname,
-             const algo::GeneratorDefinitionAlgorithm::BusGenMap& busesWithDynamicModel,
-             dfl::inputs::Configuration::StartingPointMode startingPointMode);
+  void write(boost::shared_ptr<parameters::ParametersSetCollection>& paramSetCollection, ActivePowerCompensation activePowerCompensation,
```
Comment (rosiereflo) [inline sources/Outputs/include/ParGenerator.h:55]: I hate automatic application of clang-format. There I said it :)

File: sources/Outputs/include/ParGenerator.h [hunk 3]
```diff
@@ -145,6 +134,25 @@ class ParGenerator {
   boost::shared_ptr<parameters::ParametersSet> writeGenerator(const algo::GeneratorDefinition& def, const std::string& basename,
                                                               const boost::filesystem::path& dirname);
 
+  /**
+   * @brief determine value of kGover based if a generator has a targetP equal to 0
```
Comment (rosiereflo) [inline sources/Outputs/include/ParGenerator.h:138]: a boolean named fixedP is not clear enough name in my humble opinion. Why not directly taking as parameter the targetP as double and testing if its equal to 0 inside the method?

File: sources/Outputs/include/ParGenerator.h [hunk 4]
```diff
@@ -145,6 +134,25 @@ class ParGenerator {
...
+   *
+   * @param fixedP boolean to determine if the set represents a generator with a targetP equal to 0
+   * @return 0 if fixeP is True, 1.0 otherwise
```
Comment (rosiereflo) [inline sources/Outputs/include/ParGenerator.h:141]: The values of the constant should not be given in the description

File: sources/Outputs/include/ParGenerator.h [hunk 5]
```diff
@@ -145,6 +134,25 @@ class ParGenerator {
...
+   */
+  inline double getKGoverValue(bool fixedP) {
+    return fixedP ? constants::kGoverNullValue_ : constants::kGoverDefaultValue_;
+  }
+
+  /**
+   * @brief upadte a parameter set with information specific to remote voltage regulation for a generator
```
Comment (rosiereflo) [inline sources/Outputs/include/ParGenerator.h:148]: typo: update

**Review by rosiereflo — CHANGES_REQUESTED** [review]

File: sources/Outputs/include/ParGenerator.h (hunk 1 above)
Comment (marcochiaramello) [reply sources/Outputs/include/ParGenerator.h:36]: thx!

File: sources/Outputs/include/ParGenerator.h (hunk 3 above)
Comment (marcochiaramello) [reply sources/Outputs/include/ParGenerator.h:138]: I have changed it to take targetP as parameter, thx!

File: sources/Outputs/include/ParGenerator.h (hunk 5 above)
Comment (marcochiaramello) [reply sources/Outputs/include/ParGenerator.h:148]: thx

File: sources/Algo/include/GeneratorDefinitionAlgorithm.h [hunk 6]
```diff
@@ -208,7 +217,7 @@ class GeneratorDefinitionAlgorithm {
   BusGenMap& busesWithDynamicModel_;                        ///< map of bus ids to a generator that regulates them
   const inputs::NetworkManager::BusMapRegulating& busMap_;  ///< mapping of busId and the number of generators that regulates them
   bool useInfiniteReactivelimits_;                          ///< determine if infinite reactive limits are used,
-  double tfoVoltageLevel_;                                  ///< Maximum voltage level for which we assume that generator's transformers are already described in the static description
+  double tfoVoltageLevel_;  ///< Maximum voltage level for which we assume that generator's transformers are already described in the static description
```
Comment (cossartque) [inline sources/Algo/include/GeneratorDefinitionAlgorithm.h:220]: Shouldn't it stay aligned ?

File: tests/main/reference/hvdc_HvdcPQPropDangling_diagrams/outputIIDM.xml [hunk 7]
```diff
@@ -3,35 +3,35 @@
...
     </iidm:substation>
     <iidm:substation id="_BUS___10_SS" name="BUS   10_SS" country="AF">
         <iidm:voltageLevel id="_BUS___10_VL" name="BUS   10_VL" nominalV="400" topologyKind="BUS_BREAKER">
             <iidm:busBreakerTopology>
-                <iidm:bus id="_BUS___10_TN" v="400" angle="-0.0053719764224105055"/>
+                <iidm:bus id="_BUS___10_TN" v="399.99999999999994" angle="-0.0053721187594302812"/>
             </iidm:busBreakerTopology>
             <iidm:load id="_Load__10" name="LOAD  10" loadType="UNDEFINED" p0="500" q0="0" bus="_BUS___10_TN" connectableBus="_BUS___10_TN" p="500.02884066975162" q="0"/>
-            <iidm:vscConverterStation id="VSCStation1" name="VSCS1" voltageRegulatorOn="true" lossFactor="0" voltageSetpoint="400" reactivePowerSetpoint="0" bus="_BUS___10_TN" connectableBus="_BUS___10_TN" p="-100" q="-0.011367632255344354">
+            <iidm:vscConverterStation id="VSCStation1" name="VSCS1" voltageRegulatorOn="true" lossFactor="0" voltageSetpoint="400" reactivePowerSetpoint="0" bus="_BUS___10_TN" connectableBus="_BUS___10_TN" p="-100" q="-0.031232689475746579">
```
Comment (cossartque) [inline tests/main/reference/hvdc_HvdcPQPropDangling_diagrams/outputIIDM.xml:27]: How do we explain these little changes ?

**Review by rosiereflo — APPROVED** [review]

File: sources/Algo/include/GeneratorDefinitionAlgorithm.h (hunk 6 above)
Comment (marcochiaramello) [reply sources/Algo/include/GeneratorDefinitionAlgorithm.h:220]: done

**Review by cossartque — APPROVED** [review]

**Pull request merged**